### DISCUSSION TITLE
10. How do you transform the traditional Statistical Programmer/Analyst in Pharma who primarily codes in SAS - into the future Data Scientist with multiple tools, object oriented, code review, GitHub and git versioning, software development, agile, etc.?

### DISCUSSION THREAD

**Discussion by MichaelRimler** [phuse-org/OSTCDA#10] [General]
For example:

- Provision of user training for the language
- Provision of user training for the open source culture/mindset (code review, code sharing)
- Acquisition of software development skills

**Comment by DrLynTaylor** [comment]
It's a challenging prospect and extremely scary for those of us 20+ years of SAS programming!   I can highly recommend Andy Nicholl's PSI course (https://psiweb.org/events/event-item/2024/01/29/default-calendar/psi-training-course-r-for-clinical-trial-statisticians) and also Christina Fillmore's GitHub training (https://www.youtube.com/watch?v=qJWoNcbu98M&t=198s&pp=ygUlY2hyaXN0aW5hIGZpbGxtb3JlIHIgaW4gcGhhcm1hIGdpdGh1Yg%3D%3D).     I think what draws people to R, are all the fun things you can do with it and how it can save you substantial time.   Once you start writing powerpoint presentations (that update results directly when you get new data), creating shiny apps, and designing your own web pages, it's very addictive and once people experience that, they will want to work through the pain of learning a new language  !

**Comment by randomdigits** [comment]
The transition of statistical programmers from SAS to open-source tools is a critical issue facing many organisations. When considering this challenge at the organisational level, several key components to take into account are:
- Institutional logics and their influence on the culture of the field of statistical programming (with a view to changing them)
- Organisational culture and senior management's support for transitioning to open-source tools
- Individual motivation and the impact of changing from a familiar tool with "20+ years" of expertise entangled with a sense of personal identity
It seems that this is a particularly gnarly change management challenge, in which factors beyond training and technology will disproportionately affect the outcome.

**Comment by james-eda** [comment]
To transform the traditional Statistical Programmer/Analyst in Pharma who primarily codes in SAS we introduced multilingual interactive development environments (IDEs). These environments let our programmers use multiple languages within the same editor. They select the language they prefer whenever they want, without feeling pressured to drop the most efficient tool (the one they already know best). As a result, our programmers explore alternative languages without facing significant barriers to entry. And, our programmer tell us what works best!

**Reply by zhonggai** [reply]
what kind of multilingual IDEs you have used/adopted?

**Comment by bcarancibia** [comment]
This question needs to be broken into two parts, 1) how do you transform the traditional Statistical Programming into the future Data Scientist - focused on open-source technologies and then 2) how do you add all the different tools to your toolkit. 
To facilitate this transition, organizations should encourage exploration of open-source languages and provide robust support. However, the challenge lies in balancing learning with deliverable creation. 

Training and learning are distinct; training often lacks practical application, while learning involves applying knowledge in various scenarios. Hence, the focus should shift from training to support and learning. 

Large-scale training may initially be beneficial, but its effectiveness diminishes over time due to the gap between training and application. For us, it was around 12-18 months between training and application on a deliverable. This resulted in users having to redo training on the fly while trying to deliver outputs. Our experience at GSK led us to prioritize support and mentoring over large training classes. Our AccelerateR team, a small group of open-source experts, works closely with study teams for 8-12 weeks, providing tailored training, answering queries, reviewing code, and connecting teams facing similar issues. This approach introduces concepts like code review, GitHub, and software development methods as part of the process, eliminating the need for explicit training.

Adding new tools to the toolkit should only occur once users are comfortable with open-source languages. As users delve deeper into tasks, they can gradually incorporate technical tools. The challenge is creating space to learn these tools amidst the pressure to deliver quickly. 

This transition also requires a mindset shift, as open-source languages and output development demand continuous learning. I think that this mind shift is the biggest aspect when trying to transform the traditional stats programmer/analyst.
